Compilation errors in CustomerDetails.g.cs caused by x:Bind

I can't build the project using Visual Studio 2017 15.9 Preview 3 & SDK 17763. 

The errors are:
```
3>------ Build started: Project: Inventory.App, Configuration: Debug x86 ------
3>C:\Users\habakker\Source\Repos\InventorySample\src\Inventory.App\Controls\Buttons\RoundButton.cs(44,29,44,41): warning CS0108: 'RoundButton.CornerRadius' hides inherited member 'Control.CornerRadius'. Use the new keyword if hiding was intended.
3>C:\Users\habakker\Source\Repos\InventorySample\src\Inventory.App\Controls\Buttons\RoundButton.cs(50,51,50,71): warning CS0108: 'RoundButton.CornerRadiusProperty' hides inherited member 'Control.CornerRadiusProperty'. Use the new keyword if hiding was intended.
3>C:\Users\habakker\Source\Repos\InventorySample\src\Inventory.App\obj\x86\Debug\Views\Customer\Details\CustomerDetails.g.cs(924,72,924,87): error CS0029: Cannot implicitly convert type 'string' to 'int?'
3>C:\Users\habakker\Source\Repos\InventorySample\src\Inventory.App\obj\x86\Debug\Views\Customer\Details\CustomerDetails.g.cs(950,71,950,86): error CS0029: Cannot implicitly convert type 'string' to 'decimal?'
```

The offending parts are:
```csharp
private void UpdateTwoWay_17_Text()
{
    if (this.initialized)
    {
        if (this.dataRoot != null)
        {
            if (this.dataRoot.EditableItem != null)
            {
                this.dataRoot.EditableItem.TotalChildren = this.obj17.Text; //this line
            }
        }
    }
}
```

and
```csharp
private void UpdateTwoWay_19_Text()
{
    if (this.initialized)
    {
        if (this.dataRoot != null)
        {
            if (this.dataRoot.EditableItem != null)
            {
                this.dataRoot.EditableItem.YearlyIncome = this.obj19.Text; //and this line
            }
        }
    }
}
```

The error seems to be in the `x:Bind` generated code which seems not process the `DataType="Decimal"` and `DataType="Integer"` attribute correctly. This attribute is used at https://github.com/Microsoft/InventorySample/blob/master/src/Inventory.App/Views/Customer/Details/CustomerDetails.xaml#L39,L43

--- a/src/Inventory.App/Converters/BindingTypeConverter.cs
+++ b/src/Inventory.App/Converters/BindingTypeConverter.cs
@@ -1,0 +1,30 @@
+﻿using System;
+using Windows.UI.Xaml.Data;
+
+namespace Inventory.Converters
+{
+    class BindingTypeConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            return System.Convert.ToString(value);
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, string language)
+        {
+            var t = targetType;
+
+            if (t.IsGenericType && t.GetGenericTypeDefinition().Equals(typeof(Nullable<>)))
+            {
+                if (value == null)
+                {
+                    return null;
+                }
+
+                t = Nullable.GetUnderlyingType(t);
+            }
+
+            return System.Convert.ChangeType(value, t);
+        }
+    }
+}

--- a/src/Inventory.App/Inventory.App.csproj
+++ b/src/Inventory.App/Inventory.App.csproj
@@ -140,6 +140,7 @@
     <Compile Include="Controls\Toolbars\ToolbarCommon.cs" />
     <Compile Include="Controls\WindowTitle\WindowTitle.cs" />
     <Compile Include="Controls\WrapPanel\WrapPanel.cs" />
+    <Compile Include="Converters\BindingTypeConverter.cs" />
     <Compile Include="Converters\BoolNegationConverter.cs" />
     <Compile Include="Converters\BoolToVisibilityConverter.cs" />
     <Compile Include="Converters\DateTimeFormatConverter.cs" />

--- a/src/Inventory.App/Views/Customer/Details/CustomerDetails.xaml
+++ b/src/Inventory.App/Views/Customer/Details/CustomerDetails.xaml
@@ -7,9 +7,11 @@
     xmlns:controls="using:Inventory.Controls"
     xmlns:views="using:Inventory.Views"
     xmlns:viewmodels="using:Inventory.ViewModels"
+    xmlns:converters="using:Inventory.Converters"
     mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="600">
 
     <UserControl.Resources>
+        <converters:BindingTypeConverter x:Key="BindingTypeConverter"/>
         <DataTemplate x:Key="DetailsTemplate" x:DataType="viewmodels:CustomerDetailsViewModel">
             <controls:FluidGrid Columns="2" ColumnSpacing="12" Margin="12,18,6,6" RowSpacing="12">
                 <TextBlock Style="{StaticResource FormsSection}" controls:FluidGrid.ColumnSpan="2" Text="Personal Information"/>
@@ -36,11 +38,11 @@
                 <controls:FormTextBox Header="Education" Text="{x:Bind EditableItem.Education, Mode=TwoWay}" />
                 <controls:FluidGrid Columns="2" ColumnSpacing="6" RowSpacing="12" MinColumnWidth="80">
                     <controls:FormTextBox Header="Occupation" Text="{x:Bind EditableItem.Occupation, Mode=TwoWay}" />
-                    <controls:FormTextBox Header="Yearly Income" Text="{x:Bind EditableItem.YearlyIncome, Mode=TwoWay}" DataType="Decimal" />
+                    <controls:FormTextBox Header="Yearly Income" Text="{x:Bind EditableItem.YearlyIncome, Mode=TwoWay, Converter={StaticResource BindingTypeConverter}}" DataType="Decimal" />
                 </controls:FluidGrid>
                 <controls:FluidGrid Columns="2" ColumnSpacing="6" RowSpacing="12" MinColumnWidth="80">
                     <controls:FormTextBox Header="Marital Status" Text="{x:Bind EditableItem.MaritalStatus, Mode=TwoWay}" />
-                    <controls:FormTextBox Header="Total Children" Text="{x:Bind EditableItem.TotalChildren, Mode=TwoWay}" DataType="Integer" />
+                    <controls:FormTextBox Header="Total Children" Text="{x:Bind EditableItem.TotalChildren, Mode=TwoWay, Converter={StaticResource BindingTypeConverter}}" DataType="Integer" />
                 </controls:FluidGrid>
             </controls:FluidGrid>
         </DataTemplate>

